gammapy/cube/fit.py

```diff
@@ -263,8 +263,15 @@ def _energy_axis(self):
 
     @property
     def data_shape(self):
-        """Shape of the counts data (tuple)"""
-        return self.counts.data.shape
+        """Shape of the counts or background data (tuple)"""
+        if self.counts:
+            return self.counts.data.shape
+        elif self.background_model:
+            return self.background_model.map.data.shape
+        elif self.exposure:
```

Comment:
Normally to produce the simulation you should have a background map, no? 
One should avoid using `exposure` since it is a different data space.

Reply:
Yes I completely agree. It was my confusion as well.
I do not know if it is any use case to simulate counts without a background (this is how the spectral simulation tutorial starts, so I thought there might possibly be some use case that I can't foresee). I kept the exposure only for that reason.

I am happy to remove it!

Reply:
Also, the `Mapdataset._geom()` follows this